homeassistant/components/zwave_js/migrate.py

```diff
@@ -16,8 +23,87 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class ValueID:
+    """Class to represent a Value ID."""
+
+    command_class: str
+    endpoint: str
+    property_: str
+    property_key: str | None = None
+
+    @staticmethod
+    def from_unique_id(unique_id: str) -> ValueID:
+        """
+        Get a ValueID from a unique ID.
+
+        This also works for Notification CC Binary Sensors which have their own unique ID
+        format.
+        """
+        return ValueID.from_string_id(unique_id.split(".")[1])
+
+    @staticmethod
+    def from_string_id(value_id_str: str) -> ValueID:
+        """Get a ValueID from a string representation of the value ID."""
+        parts = value_id_str.split("-")
+        property_key = parts[4] if len(parts) > 4 else None
+        return ValueID(parts[1], parts[2], parts[3], property_key=property_key)
+
+    def is_same_value_different_endpoints(self, other: ValueID) -> bool:
```

Comment:
We can just so the function makes sense, but the migration logic is such that it won't change the logic because we return early if we find an existing entry for the discovered value